### DISCUSSION TITLE
Fuse.js configuration missing?

### DISCUSSION THREAD

**Discussion by FuadEfendi** [razonyang/hugo-theme-bootstrap#720] [Q&A] [answered]
I was trying to tune Fuse.js parameters and nothing worked... 

Search for "abc" returns the same number of results without any dependency on minMatchCharLength being 1 or 3.
Setting findAllMatches=true/false doesn't impact neither.

**Comment by FuadEfendi** [comment]
Perhaps some parameters are hardcoded, and some lowercased?
```
{{- $options := dict "ignoreLocation" true "includeMatches" true "includeScore" true "keys" $keys -}}
{{- $optionWhitelist := slice "ignoreLocation" "isCaseSensitive" "minMatchCharLength" "threshold" "distance" "useExtendedSearch" -}}
{{- with .Site.Params.search.fuse -}}
  {{- range $key, $value := . -}}
    {{- range $optionWhitelist -}}
      {{- if eq $key (lower .) -}}
```

**Reply by razonyang** [reply]
Yep, it's hard code, some options are required to be set to a fixed value. 

The Hugo parameters are case-insensitive(always lowercase), so there is a map to compare them.

I'll add more parameters to the whitelist later.

**Reply by FuadEfendi** [reply]
Ok at least I found now why `distance` and `threshold` don't work: hardcoded `"ignoreLocation": true`

**Comment by razonyang** [comment, marked as answer]
Done via #722

The following options are available now.

https://github.com/razonyang/hugo-theme-bootstrap/blob/ce758a22f946e9ff04efb1cae408db69ef0e9186/exampleSite/config/_default/params.toml#L160-L169

You could check the final Fuse.js options via console(verbose mode).

![image](https://user-images.githubusercontent.com/17720932/190919922-1648482c-1287-429b-ac9b-7fc2f21e715d.png)

**Reply by FuadEfendi** [reply]
Thank you!

**Reply by FuadEfendi** [reply]
Sorry from dumb question, but my console looks different... should I enable "debug" somehow?

<img width="857" alt="image" src="https://user-images.githubusercontent.com/141863/190922135-45abccc1-5854-46aa-9f22-5ccce086a098.png">

**Reply by razonyang** [reply]
@FuadEfendi That info only present in the search page.

**Reply by FuadEfendi** [reply]
I am in "Search Results" page, clicking hard "refresh" to refresh search results

**Reply by razonyang** [reply]
That's weird, do you upgrade to the latest commit?
I also check the live site https://hbs.razonyang.com/v1/en/search/?q=markdown, seems fine.

**Reply by FuadEfendi** [reply]
Oh sorry it is shown now; after upgrading the theme

**Comment by FuadEfendi** [comment]
I suggest to use this style of "local" scratch (for "search" template),

```
{{- $scratch := newScratch -}}
{{- $scratch.Add "pages" slice -}}
{{- $scratch.Add "authors" slice }}
{{- $scratch.Add "categories" slice }}
```

I am trying to create template to generate JSON for Algolia, as in https://forestry.io/blog/search-with-algolia-in-hugo/
- but I am not sure why it doesn't filter pages at all; using this: `{{ range where .Pages "Section" "abc" }}`
I have only 5 pages in "content/abc" folder, but index size and content of "algonia.json" is exactly same as "index.json"

**Reply by razonyang** [reply]
1. Sounds good to avoid conflicting with users' global variables.

1. I don't know what source code your template is, just give you a hint, the output JSON is generated by following code. 
https://github.com/razonyang/hugo-theme-bootstrap/blob/ce758a22f946e9ff04efb1cae408db69ef0e9186/layouts/partials/search/index.json#L57-L64

You can replace it with the following to determine if it works or not.

```
{{- dict
  "hello" "world"
 | jsonify 
-}}
```

> You may need to restart the Hugo server, in my experience, the JSON file won't be regenerated sometime even you've modified the template.

**Reply by FuadEfendi** [reply]
Thank you @razonyang for support!
I started studying in depths Hugo; especially this https://www.regisphilibert.com/blog/2017/04/hugo-scratch-explained-variable/
